### ISSUE TITLE
Review of Optional and Informative Tests

### ISSUE THREAD

**Issue by valentindey** [secvisogram/csaf-validator-lib#13]
Here's a list of issues I found reviewing the already checked in optional and informative tests. For tests not mentioned, I could not find any issues.

**Optional Tests**

* [Test 6.2.1](https://github.com/secvisogram/csaf-validator-lib/blob/main/lib/optionalTests/optionalTest_6_2_1.js)
  - Is the `?? false` part necessary for all the checks? If so, I'd like to understand better why. I'm referring to [these lines](https://github.com/secvisogram/csaf-validator-lib/blob/main/lib/optionalTests/optionalTest_6_2_1.js#L263-L271).

* [Test 6.2.10](https://github.com/secvisogram/csaf-validator-lib/blob/main/lib/optionalTests/optionalTest_6_2_10.js)
  - The test checks if the tlp label is present but it does not check for validity of the used label, i.e. if they are one of the allowed enum values `AMBER`, `GREEN`, `RED`, or `WHITE`.

* [Test 6.2.11](https://github.com/secvisogram/csaf-validator-lib/blob/main/lib/optionalTests/optionalTest_6_2_11.js)
  - The test does not check whether the url starts with `https://`.
  - It does falsely replace `-` characters for checking the string.
  - The specification has changed and now replaces sequences of not allowed characters with a single underscore. The new specification also supplies a regex that can be used to the the replacement: `[^+\-a-z0-9]+`

* [Test 6.2.12](https://github.com/secvisogram/csaf-validator-lib/blob/main/lib/optionalTests/optionalTest_6_2_12.js)
  - Running and modifying the existing test, it seems like skipping the `lang` property completely will pass the test. If I understand the specification correctly, it should also fail if the property isn't present at all.

**Informative Tests**

* [Test 6.3.2](https://github.com/secvisogram/csaf-validator-lib/blob/main/lib/informativeTests/informativeTest_6_3_2.js)
  - The schemas for CVSS v3.0 and v3.1 require the properties `version`, `vectorString`, `baseScore`, and `baseSeverity` while here they are listed as `optionalProperties`. It's probably out of the scope of this test, but I'm wondering if we could make it conform to the schema more. The same may apply for [test 6.3.1](https://github.com/secvisogram/csaf-validator-lib/blob/main/lib/informativeTests/informativeTest_6_3_1.js).
  - An additional check for the pattern of `vectorString` could be performed to ensure v3.1 is used. Checking for the version prefix may be fine, though and this would be out of the scope of this test resp. extend it way beyond what is required by the specification.

**Comment by valentindey** [comment]
For test 6.2.11 there's another change with CSD02 that probably has an effect: https://github.com/oasis-tcs/csaf/issues/440 Maybe @tschmidtb51 can add to that, if something needs to change for the test.

**Comment by valentindey** [comment]
Tracking resolutions to the found issues in this comment:

- [x] 6.2.1 has been resolved in 2da211ed50e87533a612621c54de655af576946a and I got a nice explanation, thanks :) 
- [ ] 6.2.10 currently still open for discussion. The correct values are already checked by schema validation.
- [x] 6.2.11 has been resolved in c678a6ca756ca6c1a2b7e3985af793ed29dfef32
- [x] 6.2.12 my comment resulted from me not understanding some workings of javascript. `prop: undefined` is quasi equivalent to "deleting" the prop which needs to be done, as the test doc `minimalDoc` has a `lang` set. Thus this was not an issue.
- [x] 6.3.2 my comments most probably exceed the scope of the test as they are covered by schema validation. I consider it resolved.
